### ISSUE TITLE
3 Punches and 3 Kicks Macro does not function properly in 2 player vs games (windows), or at all in Android

### ISSUE THREAD

**Issue by prataczak** [libretro/fbalpha#90]
This is probably a very minor issue for most people, but it is the very reason I've always used FBA for Street Fighter and Marvel vs. Capcom series games. It makes the games a lot easier to pull off dashes, super jumps, and special moves. FBA has an option to configure 3 punches and 3 kicks to a button. It works fine for player 1, on the windows version of Retroarch. But if you try to configure player 2, every time player 1 hits 3 punches or 3 kicks during a 2 player vs. game, player 2 will do the same thing. So something is not right. 

Also, there appears to be 2 places to configure this from FBA's quick menu - both in options, and in controls. In the controls section, I didn't see an option to configure 3 punches and 3 kicks for player 2, only for player 1. A little confusing here too, because it seems like if I play with the 3 punches and 3 kicks settings for player 2 in the 'options' section, a '3 kicks' setting will appear in the 'controls' section for player 2, but not 3 punches. Weird.

Also, in the Android version, the 3 punches and 3 kicks functionality is completely broken, and will not work even for player 1.

**Comment by barbudreadmon** [comment]
From @AkimanBengus : 
FBA handles some macros input for certain games/systems.
Street Fighter like games: 3x Punch and 3x Kick
Neogeo: AB, BC, CD ...
PGM: Button 12, 23, 123 ...

It's now possible to choose the macro you want to activate via the core
options.
For Street Fighter only the L2, L3, R2 and R3 buttons will be available.
For the others all the L and R buttons are free to use.
By default, all the macros are set on 'None'.
When you choose a button for a macro, this macro will be activated and
will also appear in the 'Core Input Remapping Options'
The macro core options are affected by game to be able to configure
different macro for each game.

I don't know if it'll help you

**Comment by prataczak** [comment]
I have no problem setting up the macros. But if I'm trying to play a 2 player game with someone, when I use the macros on controller 1, player 1 and player 2 both do the macro.

That is the issue here.

Also, the macros do not work at all in the Android version.

**Comment by AkimanBengus** [comment]
I fixed the Player 2 Macros.
But I can't check the android version of RetroArch

**Comment by prataczak** [comment]
Thank you for that, seems to be working great now!

**Comment by AkimanBengus** [comment]
You are welcome.

For the Android problem it would be good to create a separated issue and close this one for the 2P problem :-)

**Comment by prataczak** [comment]
Ok, will do!

**Comment by barbudreadmon** [comment]
On the same principle, i have been wondering since yesterday how to have the "neogeo gamepad scheme" option work on a per player basis, any idea @AkimanBengus  ?

I also have been thinking about adding a new option to switch L/R to R1/R2, which would work the same way as the "neogeo gamepad scheme", but for capcom games (having R1/R2 for strong punch / strong kick is something common these days on capcom games). That's something i would be able to code by myself, but as said above, i'm not happy with how the current "neogeo gamepad scheme" work on a general basis.

**Comment by AkimanBengus** [comment]
I implemented the "neogeo gamepad scheme" per player :-)

For the L/R to R1/R2, I suppose than it can be implemented with the same logic.
I will look when I will have time.
I also needed this option because I have a MadCatz Gamepad and FightStick but I used the "auto remapping" of RetroArch to solve the problem.
I prefer the "auto remapping" because it solve also the problem for the snes and the megadrive
